### ISSUE TITLE
Challenge completion celebration

### ISSUE THREAD

**Issue by adamjonas** [saving-satoshi/saving-satoshi#58]
As a learner, when I complete a challenge, I should be exhilarated and encouraged by completing a challenge.

Possibilities: 
1. Modal drop on completion with congratulations with some advancement of the narrative
2. An improved version of the completion banner
<img width="1315" alt="Screen Shot 2023-01-06 at 1 09 31 PM" src="https://user-images.githubusercontent.com/755825/211073083-bf82a4f6-4852-4c8c-a0cc-9c666b892ac1.png">

**Comment by aureleoules** [comment]
Is this a duplicate of #53 ?

**Comment by adamjonas** [comment]
@GBKS any input on this one?

**Comment by GBKS** [comment]
Something we should tackle, for sure. Have not thought super deeply about it, but here are some ideas:

- The inputted text turns a celebratory green when it's correct (see [here](https://github.com/saving-satoshi/saving-satoshi/pull/97#issuecomment-1400007697))
- Change "Challenge completed" to be more informational and unique (see [here](https://www.figma.com/file/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=1056%3A14226&t=c5pqnLqCDgVfvlS4-1))
- Some kind of elaborate animation that's fun and playful (Duolingo does this well, some info [here](https://blog.duolingo.com/streak-milestone-design-animation/)). I have not looked much into animation since there has been more basic stuff to take care of

**Comment by ecurrencyhodler** [comment]
It would be great if we can explore animations. That would 10x the endorphines! If someone investigates how to integrate animations, design can help create something for eng to implement?

**Comment by GBKS** [comment]
Most of the UI animations (really transitions) we need can be handled via the React/Next built-in transition functionality. For more elaborate animations, like drawing out Elliptic Curve Cryptography math, we may need to set up [Lottie](https://airbnb.io/lottie/).

**Comment by GBKS** [comment]
I just got assigned this, so one extra comment. Before worrying about how to do this technically, let's first come up with animations. The tech solution can then follow our intent.

**Comment by ecurrencyhodler** [comment]
Do you think this is worth including into v1?

**Comment by GBKS** [comment]
Depends on how we scope it out. When this issue was posted on Jan 6, the experience looked very different. Maybe this is already addressed?

**Comment by benalleng** [comment]
I tried my hand in a little animation practice with #311. It doesn't work quite like i want it to however but still something to consider as inspiration for this

**Comment by ecurrencyhodler** [comment]
Thanks Ben!

Yea Christoph, I feel like this is a "nice to have" but not necessary. 

@adamjonas do you feel like this issue has been addressed in the demo? If not, was there something more specific you had in mind to help define it a bit more?

**Comment by adamjonas** [comment]
Let me qualify this with that I'm a 4/10 on the strength of my conviction here. And so, while the challenge completion is certainly better, I don't think that the inline green text for that first challenge completion -- the magical moment of seeing satoshi's secret message revealed -- is the kind of endorphin hit that I was imagining. An animation or modal celebration would be more encouraging I believe.

**Comment by ecurrencyhodler** [comment]
Okay. This sounds like a "nice to have." I'm going to remove this from v1 and adding it to the backlog.

Ben, if you want I'm open to using the animations in your PR if @GBKS approves. 

Moving forward let's have people leave a comment here on this issue with ideas or design proposal that design approves before submitting a PR.

**Comment by ecurrencyhodler** [comment]
Just to throw another idea on the table here for future reference, here's something we did at Casa that people seemed to really enjoy.

https://user-images.githubusercontent.com/32662508/225740639-6adbb0ba-236d-4051-95eb-8d7874cbe9c0.mov

**Comment by GBKS** [comment]
Ah, the classic confetti.

Thanks for the PR exploration, @benalleng, great to see that it's easy to get Lottie animations in there. I don't find this specific animation hitting the spot. At this moment, the user is focused on the input field, pastes the code, and it turns green. The animated check is in the bottom-left corner, out of the focus-zone. It's also unexpected because usually nothing happens in UIs when pasting content. The effect would be much stronger and intuitive if the animation is right where the user is looking. Let's do some design exploration and discuss some variations that we can ideally also work across other challenge types.

**Comment by benalleng** [comment]
Yeah, it's definitely not the best but it was free! I've been looking a little at Duolingo again and they have some fun modals to show the user's streak level. I'll try to find an example to paste here

**Comment by adamjonas** [comment]
I'm glad the exploration is getting the juices flowing but to be clear, I don't think this is critical path at the moment. I'd suggest this ticket remain open but not be at the top of the priority list.
